src/core/domain.py

```diff
@@ -37,11 +37,14 @@ class Status(Enum):
     INSUFFICIENT_CLUSTER_PERMISSIONS = BlockedStatus(
         "Insufficient cluster permissions. Try: juju trust --scope=cluster <app-name>"
     )
+    INVALID_EXPOSE_EXTERNAL = BlockedStatus("Invalid config option: expose-external")
```

Comment:
**suggestion, out of scope** I would put a "general" status and message for when config option validation fails. Otherwise, as we increase the number of config options also the statuses would increase. I believe the pydantic messages are fairly nice, and we could just streamline the validation error that we get from pydantic. 

I understand that this may require to rework the use of the enum we have here, and it is more complicated that just one line change. So I would say that it is out of scope (and the solution here is good enought for now), but maybe we should have a thought on how we can change it in the future to make it more scalable.

Reply:
@deusebio I agree with your point.

This line is a leftover after I have removed explicit validation of `expose-external` from our charm.

One thing I noticed with the structured_config and thus subsequently the data_models lib is that the lib doesn't set the charm to "Blocked" state automatically when the config validation fails. And since the pydantic's ValidationError is just raised by the charm thus crashing the charm whenever charm.config is accessed, the try / except and set_block_status() should be done at the earliest level of charm.py itself and not in the compute_status decorator as we have desired for central place of setting app status.

For the scope of this PR, I have left the validation entirely to structured_config and thus the charm will crash when expose-external is not one of permitted values. It would be great if its possible to refactor the `data_models` library a little bit so that it can set a generic "BlockedStatus" by itself when `pydantic` raises a `ValidationError`.